starter-code/src/memory.js

```diff
@@ -1,15 +1,54 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function (cards) {
+  var self = this;
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+   self.cards = cards;
+   self.pickedCards = [];
+   self.pairsClicked = 0;
+   self.pairsGuessed = 0;
+};
 
-// };
+ MemoryGame.prototype.shuffleCard = function (cardsArr) {
+  var self = this;
 
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+  newArray = [];
 
-// }
+  for (var i = 0; i < cardsArr.length; i++){
+    var randomIndex = Math.floor(Math.random() * cardsArr.length); 
+    newArray.push(cardsArr[randomIndex]);
```

Comment:
Take care!
Here you hare pushing random elements to newArray from cardsArr. That means that maybe you will push some element more than one time.